mozilla-release/mobile/android/app/src/photon/res/layout/two_line_page_row.xml

```diff
@@ -2,84 +2,95 @@
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_marginBottom="2dp"
-            android:background="@null"
-            android:lineSpacingMultiplier="1.1"
-            android:textColor="@color/two_line_page_row_title"
-            android:textSize="15sp"
-            gecko:fadeWidth="90dp"
-            tools:text="This is a long test title"/>
+        <ImageView
+            android:id="@+id/icon"
+            android:layout_width="@dimen/ghostery_history_icon_width"
+            android:layout_height="@dimen/ghostery_history_icon_height"
+            android:layout_gravity="center_vertical"
+            android:layout_marginLeft="20dp"
+            android:layout_marginRight="20dp"
+            tools:background="@android:color/holo_green_light" />
 
-        <org.mozilla.gecko.widget.themed.ThemedTextView
-            android:id="@+id/url"
-            style="@style/Widget.TwoLinePageRow.Url"
-            android:layout_width="match_parent"
+        <LinearLayout
+            android:layout_width="0dp"
             android:layout_height="wrap_content"
-            android:background="@null"
-            android:drawablePadding="5dp"
-            android:lineSpacingMultiplier="1.2"
-            android:maxLength="1024"
-            android:textColor="@color/two_line_page_row_url"
-            android:textSize="12sp"
-            gecko:fadeWidth="90dp"
-            tools:drawableLeft="@drawable/ic_url_bar_tab"
-            tools:drawableStart="@drawable/ic_url_bar_tab"
-            android:drawableTint="@android:color/white"
-            tools:text="http://test.com/test"/>
-        <!-- Cliqz end -->
+            android:layout_gravity="center_vertical"
+            android:layout_weight="1"
+            android:orientation="vertical"
+            android:paddingEnd="10dp"
+            android:paddingRight="10dp">
 
-    </LinearLayout>
+            <org.mozilla.gecko.widget.themed.ThemedTextView
+                android:id="@+id/title"
+                style="@style/Widget.TwoLinePageRow.Title"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginBottom="2dp"
+                android:background="@null"
+                android:lineSpacingMultiplier="1.1"
+                android:textColor="@color/two_line_page_row_title"
+                android:textSize="15sp"
+                gecko:fadeWidth="90dp"
+                tools:text="This is a long test title" />
+
+            <org.mozilla.gecko.widget.themed.ThemedTextView
+                android:id="@+id/url"
+                style="@style/Widget.TwoLinePageRow.Url"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:background="@null"
+                android:drawablePadding="5dp"
+                android:drawableTint="@android:color/white"
```

Comment:
this applied for >= 21 only, did you handle it before or not needed

Reply:
I handled it in the code. It should work on >=21 now